core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -80,11 +80,10 @@ impl ReceiveAndBuffer for SanitizedTransactionReceiveAndBuffer {
         count_metrics: &mut SchedulerCountMetrics,
         decision: &BufferedPacketsDecision,
     ) -> Result<usize, ()> {
-        let remaining_queue_capacity = container.remaining_capacity();
-
+        const MAX_RECEIVE_PACKETS: usize = 5_000;
         const MAX_PACKET_RECEIVE_TIME: Duration = Duration::from_millis(10);
         let (recv_timeout, should_buffer) = match decision {
-            BufferedPacketsDecision::Consume(_) => (
+            BufferedPacketsDecision::Consume(_) | BufferedPacketsDecision::Hold => (
```

Comment:
Behavioral change: `Hold` can now result in 0 timeout if the container is not empty. This is intended to avoid long timeouts of 10ms if a Hold pops up between our leader slots (it can)

Reply:
In the profiles I've been taking so far, we seem to spend exactly 20ms doing nothing in between slots. Curious to see what happens now.

Reply:
didn't you recently profile on the box with this change in there? did you see something there?